vllm_spyre/worker/spyre_embedding_model_runner.py

```diff
@@ -42,11 +42,20 @@ def __init__(
                          is_driver_worker=is_driver_worker)
 
         pooler_config = model_config.pooler_config
-        self.pooler = Pooler.from_config_with_defaults(
-            pooler_config,
-            pooling_type=PoolingType.CLS,
-            normalize=True,
-            softmax=False)
+        if hasattr(Pooler, "from_config_with_defaults"):
+            # TODO: remove this when we no longer support
+            # vllm version v0.9.2
+            self.pooler = Pooler.from_config_with_defaults(
+                pooler_config,
+                pooling_type=PoolingType.CLS,
+                normalize=True,
+                softmax=False)
+        else:
+            self.pooler = Pooler.for_embed(
+                pooler_config=pooler_config,
+                default_pooling_type=PoolingType.CLS,
+                default_normalize=True,
+                default_softmax=False)
```

Comment:
@maxdebayser I think just need this piece of code from your PR to enable vllm:main breaking changes, I've removed all pooling changes from this PR. I'm fine in waiting for your PR to get in first in which case I'll rebase and remove this change